tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -4,79 +4,226 @@
 
 
-def test_1vcpu(test_microvm_with_ssh, network_config):
-    """Test CPU feature emulation with 1 vCPU."""
-    test_microvm = test_microvm_with_ssh
-    test_microvm.spawn()
+class CpuVendor(Enum):
+    """CPU vendors enum."""
 
-    # Set up the microVM with 1 vCPUs, 256 MiB of RAM, no network ifaces, and
-    # a root file system with the rw permission. The network interfaces is
-    # added after we get a unique MAC and IP.
-    test_microvm.basic_config(vcpu_count=1)
+    AMD = auto()
+    INTEL = auto()
 
-    _tap, _, _ = test_microvm.ssh_network_config(network_config, '1')
-    test_microvm.start()
-    expected_cpu_topology = {
-        "CPU(s)": "1",
-        "On-line CPU(s) list": "0",
-        "Thread(s) per core": "1",
-        "Core(s) per socket": "1",
-        "Socket(s)": "1",
-        "NUMA node(s)": "1"
-    }
-    _check_cpu_topology(test_microvm, expected_cpu_topology)
 
+def _setup_microvm(test_microvm_with_ssh, network_config, vcpu_count,
+                   ht_enabled):
+    test_microvm_with_ssh.spawn()
+    test_microvm_with_ssh.basic_config(vcpu_count=vcpu_count,
+                                       ht_enabled=ht_enabled)
+    test_microvm_with_ssh.ssh_network_config(network_config, '1')
+    test_microvm_with_ssh.start()
 
-def test_2vcpu_ht_disabled(test_microvm_with_ssh, network_config):
-    """Test CPU feature emulation with 2 vCPUs, and no hyperthreading."""
-    test_microvm = test_microvm_with_ssh
-    test_microvm.spawn()
-
-    # Set up the microVM with 2 vCPUs, 256 MiB of RAM, 0 network ifaces, and
-    # a root file system with the rw permission. The network interfaces is
-    # added after we get a unique MAC and IP.
-    test_microvm.basic_config(vcpu_count=2, ht_enabled=False)
-
-    _tap, _, _ = test_microvm.ssh_network_config(network_config, '1')
 
-    test_microvm.start()
-
-    expected_cpu_topology = {
-        "CPU(s)": "2",
-        "On-line CPU(s) list": "0,1",
-        "Thread(s) per core": "1",
-        "Core(s) per socket": "2",
-        "Socket(s)": "1",
-        "NUMA node(s)": "1"
-    }
-    _check_cpu_topology(test_microvm, expected_cpu_topology)
+def _get_cpu_vendor():
+    cif = open('/proc/cpuinfo', 'r')
+    host_vendor_id = None
+    while True:
+        line = cif.readline()
+        if line == '':
+            break
+        mo = re.search("^vendor_id\\s+:\\s+(.+)$", line)
+        if mo:
+            host_vendor_id = mo.group(1)
+    cif.close()
+    assert host_vendor_id is not None
 
+    if host_vendor_id == "AuthenticAMD":
+        return CpuVendor.AMD
+    return CpuVendor.INTEL
 
-def _check_cpu_topology(test_microvm, expected_cpu_topology):
-    """Perform common microvm setup for different CPU topology tests.
 
-    This is a wrapper function for calling lscpu and checking if the
-    command returns the expected cpu topology.
-    """
+def _check_guest_cmd_output(test_microvm, guest_cmd, expected_header,
```

Comment:
nit: I'd love to see this helper function in a `utils` file. There are several similar implementations scattered throughout our CI code-base.

Reply:
Just moving it in a `utils` file is not enough. We should also change the old code in order to call the `utils` method.

I created #1068 in order to deal with this issue as "technical debt"